osu.Game/Rulesets/UI/Scrolling/Algorithms/OverlappingScrollAlgorithm.cs

```diff
@@ -76,21 +73,6 @@ public void Reset()
         /// </summary>
         /// <param name="time">The time which the <see cref="MultiplierControlPoint"/> should affect.</param>
         /// <returns>The <see cref="MultiplierControlPoint"/>.</returns>
-        private MultiplierControlPoint controlPointAt(double time)
-        {
-            if (controlPoints.Count == 0)
-                return new MultiplierControlPoint(double.NegativeInfinity);
-
-            if (time < controlPoints[0].StartTime)
-                return controlPoints[0];
-
-            searchPoint.StartTime = time;
-            int index = controlPoints.BinarySearch(searchPoint);
-
-            if (index < 0)
-                index = ~index - 1;
-
-            return controlPoints[index];
-        }
+        private MultiplierControlPoint controlPointAt(double time) => ControlPointInfo.BinarySearch(controlPoints, time) ?? new MultiplierControlPoint(double.NegativeInfinity);
```

Comment:
I think this attempt of re-using `ControlPointInfo.BinarySearch()` in fact changes behaviour?

`ControlPointInfo.BinarySearch()` [returns null if `time` given is before the first control point](https://github.com/ppy/osu/blob/d237c818f6d744f4a92697f7b45cbc6ff7747dc4/osu.Game/Beatmaps/ControlPoints/ControlPointInfo.cs#L220-L221), which then would cause this new code to fall back to the `new MultiplierControlPoint(double.NegativeInfinity)`.

Meanwhile the old code [would use the first multiplier point](https://github.com/ppy/osu/blob/31f499a9502728ff0855564410324681853ec5b3/osu.Game/Rulesets/UI/Scrolling/Algorithms/OverlappingScrollAlgorithm.cs#L84-L85) instead.

Reply:
Good catch. I've fixed this and added better commentary.